pkg/e2e/framework/framework.go

```diff
@@ -59,83 +59,122 @@ func init() {
 
+	flag.StringVar(&proxySSHUser, "proxy-ssh-user", "root", "Ssh user for proxy")
+	flag.StringVar(&proxySSHKey, "proxy-ssh-key", "", "Ssh key for proxy")
+	flag.StringVar(&proxySSHHost, "proxy-ssh-host", "", "Ssh host for proxy")
+
 	flag.Parse()
 }
 
+type ErrNotExpectedFnc func(error)
+type ByFnc func(string)
+
+type SSHConfig struct {
+	Key  string
+	User string
+	Host string
+}
+
 // Framework supports common operations used by tests
 type Framework struct {
 	KubeClient            *kubernetes.Clientset
 	CAPIClient            *clientset.Clientset
 	APIRegistrationClient *apiregistrationclientset.Clientset
 	Kubeconfig            string
 	RestConfig            *rest.Config
-	SSHKey                string
-	SSHUser               string
-	ActuatorImage         string
+
+	SSH *SSHConfig
+
+	ActuatorImage  string
+	ErrNotExpected ErrNotExpectedFnc
+	By             ByFnc
 }
 
 // NewFramework setups a new framework
 func NewFramework() *Framework {
 	if sshkey == "" {
-		panic("-sskey not set")
+		panic("-sshkey not set")
 	}
 	if kubeconfig == "" {
 		panic("-kubeconfig not set")
 	}
 	f := &Framework{
-		Kubeconfig:    kubeconfig,
-		SSHKey:        sshkey,
-		SSHUser:       sshuser,
+		Kubeconfig: kubeconfig,
+		SSH: &SSHConfig{
+			Key:  sshkey,
+			User: sshuser,
+		},
+
 		ActuatorImage: actuatorImage,
 	}
 
+	f.ErrNotExpected = f.DefaultErrNotExpected
+	f.By = f.DefaultBy
+
 	BeforeEach(f.BeforeEach)
 	return f
 }
 
-func NewFrameworkFromConfig(config *rest.Config) *Framework {
+func DefaultSSHConfig() (*SSHConfig, error) {
 	if sshkey == "" {
-		panic("-sskey not set")
+		return nil, fmt.Errorf("-sshkey not set")
 	}
 
+	return &SSHConfig{
+		Key:  sshkey,
+		User: sshuser,
+	}, nil
+}
+
+func NewFrameworkFromConfig(config *rest.Config, sshConfig *SSHConfig) (*Framework, error) {
 	f := &Framework{
 		RestConfig:    config,
-		SSHKey:        sshkey,
+		SSH:           sshConfig,
 		ActuatorImage: actuatorImage,
 	}
 
-	f.buildClientsets()
-	return f
+	err := f.buildClientsets()
+	return f, err
 }
 
-func (f *Framework) buildClientsets() {
+func (f *Framework) buildClientsets() error {
 	var err error
 
-	By("Creating a kubernetes client")
 	if f.RestConfig == nil {
 		f.RestConfig, err = controller.GetConfig(f.Kubeconfig)
-		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			return nil
+		}
 	}
 
 	if f.KubeClient == nil {
 		f.KubeClient, err = kubernetes.NewForConfig(f.RestConfig)
-		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			return nil
+		}
 	}
 
 	if f.CAPIClient == nil {
 		f.CAPIClient, err = clientset.NewForConfig(f.RestConfig)
-		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			return nil
+		}
 	}
 
 	if f.APIRegistrationClient == nil {
 		f.APIRegistrationClient, err = apiregistrationclientset.NewForConfig(f.RestConfig)
-		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			return nil
+		}
 	}
+
+	return nil
```

Comment:
Success! That's why I don't understand why the others don't return err. It seems errors are silent.